airflow/models.py

```diff
@@ -3678,7 +3757,7 @@ def refresh_from_db(self, session=None):
             DR.dag_id == self.dag_id,
             DR.execution_date == self.execution_date,
             DR.run_id == self.run_id
-        ).one()
+        ).first()
```

Comment:
Do you want me to do a full revert on all of the code? :-). I consider this your code at the moment. I suggest creating a small PR that addresses this and 2 other PRs that address the bugs.